pkg/ocicni/ocicni.go

```diff
@@ -467,33 +467,34 @@ func (plugin *cniNetworkPlugin) GetPodNetworkStatus(podNetwork PodNetwork) ([]cn
 
 	results := make([]cnitypes.Result, 0)
 	if err := plugin.forEachNetwork(&podNetwork, func(network *cniNetwork, ifName string, podNetwork *PodNetwork) error {
-		version := "4"
-		ip, mac, err := getContainerDetails(plugin.nsManager, podNetwork.NetNS, ifName, "-4")
-		if err != nil {
-			ip, mac, err = getContainerDetails(plugin.nsManager, podNetwork.NetNS, ifName, "-6")
-			if err != nil {
-				return err
-			}
-			version = "6"
-		}
+		ips := make([]*cnicurrent.IPConfig, 0)
+		interfaces := make([]*cnicurrent.Interface, 0)
 
-		// Until CNI's GET request lands, construct the Result manually
-		results = append(results, &cnicurrent.Result{
-			CNIVersion: "0.3.1",
-			Interfaces: []*cnicurrent.Interface{
-				{
+		for _, version := range []string{"4", "6"} {
+			ip, mac, err := getContainerDetails(plugin.nsManager, podNetwork.NetNS, ifName, "-"+version)
+			if err == nil {
+				interfaces = append(interfaces, &cnicurrent.Interface{
```

Comment:
This cannot be correct as it will append two copies of the interface struct if the interface has both v4 and v6 addresses. Instead, there should be only one Interface and two IPs that refer to that interface by index.

What I'd do instead is something like :
```
var intf *cnicurrent.Interface
for _, version := range {"4", "6"} {
    ip, mac, err := getContainerDetails(...)
    if err == nil {
        if intf == nil {
           intf = &cnicurrent.Interface{ blah blah }
        }
        ips = append(ips, blah blah)
    }
}
if intf == nil || len(ips) == nil {
    return fmt.Errorf( blah blah lbah )
}

results = append(results, &cnicurrent.Result{
    CNVersion: "0.3.1",
    Interfaces: []*cnicurrent.Interface{intf},
    IPs: ips,
})
```

or something like that. Again, for any given 'ifname' there should be only one element in the Interfaces array.

Reply:
ok, I will modify it.

Reply:
@dcbw I have modified it, PTAL,  thanks.